src/Mcamara/LaravelLocalization/Middleware/LocaleSetLanguage.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Mcamara\LaravelLocalization\Middleware;
+
+use Closure;
+use Illuminate\Contracts\Config\Repository as ConfigRepository;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\Translation\Translator;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Mcamara\LaravelLocalization\Exceptions\SupportedLocalesNotDefined;
+use Mcamara\LaravelLocalization\Exceptions\UnsupportedLocaleException;
+use Mcamara\LaravelLocalization\LanguageNegotiator;
+use Mcamara\LaravelLocalization\LaravelLocalization;
+
+class LocaleSetLanguage extends LaravelLocalizationMiddlewareBase
+{
+    public function __construct(
+        private readonly ConfigRepository $configRepository,
+        private readonly Application $app,
+        private readonly Translator $translator,
+        private readonly LaravelLocalization $laravelLocalization,
+    ){
+    }
+
+    public function handle(Request $request, Closure $next)
+    {
+        // I am not 100% sure if we should skip setLang for
+        // ignored urls.
+        if ($this->shouldIgnore($request)) {
+            return $next($request);
+        }
+
+        $locale = $request->route('locale');
+
+        if($locale == null || empty($this->getSupportedLocales()[$locale])) {
+            $locale = $this->fallbackLocale($request);
+        }
+
+        $this->app->setLocale($locale);
+        $this->translator->setLocale($locale);
+        $this->laravelLocalization->setCurrentLocale($locale);
+
+        // Regional locale such as de_DE, so formatLocalized works in Carbon
+        $regional = $this->getLocaleRegional($locale);
+        $suffix = $this->configRepository->get('laravellocalization.utf8suffix');
+        if ($regional) {
+            setlocale(LC_TIME, $regional . $suffix);
+            setlocale(LC_MONETARY, $regional . $suffix);
+        }
+
+        return $next($request);
+    }
+
+    protected function getSupportedLocales()
+    {
+        $locales = $this->configRepository->get('laravellocalization.supportedLocales');
+
+        if (empty($locales) || !\is_array($locales)) {
+            throw new SupportedLocalesNotDefined();
+        }
+
+        return $locales;
+    }
```

Comment:
We should use it from `LaravelLocalization` class and remove it from the`middleware`. Initially, I thought it could be removed from `LaravelLocalization` later, but that’s not possible since it’s actively used and mentioned in the documentation.

Reply:
Restored